packages/react-reconciler/src/ReactFiberTracingHook.js

```diff
@@ -0,0 +1,346 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber, FiberRoot} from './ReactInternalTypes';
+
+import {enableTracingHooks} from 'shared/ReactFeatureFlags';
+import getComponentNameFromFiber from './getComponentNameFromFiber';
+
+export function markCommitStarted(root: FiberRoot): void {
+  if (enableTracingHooks) {
+    if (
+      root.tracingHooks != null &&
+      typeof root.tracingHooks.onCommitStarted === 'function'
+    ) {
+      root.tracingHooks.onCommitStarted();
+    }
+  }
+}
+
+export function markCommitStopped(root: FiberRoot): void {
+  if (enableTracingHooks) {
+    if (
+      root.tracingHooks != null &&
+      typeof root.tracingHooks.onCommitStopped === 'function'
+    ) {
+      root.tracingHooks.onCommitStopped();
+    }
+  }
+}
+
+export function markComponentRenderStarted(
+  root: ?FiberRoot,
+  workInProgress: Fiber,
+): void {
+  if (enableTracingHooks) {
+    const onComponentRenderStarted =
+      root &&
+      root.tracingHooks != null &&
+      typeof root.tracingHooks.onComponentRenderStarted === 'function'
+        ? root.tracingHooks.onComponentRenderStarted
+        : null;
+    if (onComponentRenderStarted != null) {
+      const componentName = getComponentNameFromFiber(workInProgress);
+      onComponentRenderStarted(componentName);
```

Comment:
It seems this file is created to provide APIs for tracing hooks as a separate variant to the devtools profiling hooks. Is the goal to decouple the two so that they can both be used in dev environment? Or do you plan to register these hooks (from `root.tracingHooks.onComponentRenderStarted`) somewhere in native side so that we can implement more efficient tracing logic depends on the hosting platform?

Reply:
So they can be used together, but also to avoid the performance overhead the comes from the DevTools implementation (which does more work than just calling this functions at specific times).